packages/core/connection.ts

```diff
@@ -21,13 +20,9 @@ export default class Connection {
     return this.conn.credentials.askForPassword;
   }
 
-  public connect() {
-    return this.query('SELECT 1;', true)
-      .then(() => {
-        this.connected = true;
-        return true;
-      })
-      .catch((e) => Promise.reject(e));
+  public async connect() {
+    await this.query('SELECT 1;', true);
```

Comment:
On Oracle it should be 'SELECT 1 FROM DUAL;'. Maybe ConnectionDialect should expose a testConnection method...

Reply:
Can you work on that? It makes sense for me. We could check if the connection has a method to test, otherwise we use the default.

Reply:
I don't have permision to push into this repo so I pushed to my fork https://github.com/mickeypearce/vscode-sqltools/tree/feature/oracle.

I am not super fluent in TS but I made it work. You can change / refactor - no hard feelings on my side..

Reply:
I've changed the names and the connection opening only. Looking into the libraries for pqsql, mysql and oracle, they are better managed when using connection pooling, so I just grab a connection, run the queries, then I request to close it. Next execution it should just open fine.

Also, would you like to be a contributor? I can give you access to the repository. Since I can't test oracle easily, would be nice if you could just move forward with oracle issues.

I see you have some extensions too, so you know how it works.

What do you think?

Thanks!

Reply:
Thanks. I accepted your invitation.